### PR TITLE
Docs: Code conventions improvements

### PR DESCRIPTION
**Pull request by platinumazure**
Highlights:

* File overview section: Removed notes about copyright
* Strict mode: Rewrote guidelines to favor global "strict" mode
* Other tweaks: Fixed ambiguity about for statement spacing

## File Overview

I removed everything to do with copyright from that section. Now that we are in the jQuery foundation, we have decided to just use the "author" field and no copyright or license.

## Strict Mode

Our repository [enforces global "strict"](https://github.com/eslint/eslint/blob/89580a40557093184b5e164b7ef284670d47090b/packages/eslint-config-eslint/default.yml#L94), but the code conventions imply that we should have ["strict" mode in functions](https://github.com/eslint/eslint/blob/89580a40557093184b5e164b7ef284670d47090b/docs/developer-guide/code-conventions.md#strict-mode).

My assumption is that those conventions have more to do with non-module code, whereas NodeJS actually has an extra scope, global return, etc. and so it is safe to have "use strict" globally (and possibly better, since it means we don't have sections of files that are not strict).

## Other

Whitespace notes were slightly ambiguous about the spacing between elements of the for loop initialization. I added a note specifying that whitespace should only be after semicolons, not before.

(Apologies to @nzakas for creating a branch on the main repo-- I was using the GitHub editor and apparently did something wrong. Hopefully I'll figure it out before I do another GitHub doc change.)

### CODE REVIEW

File: docs/developer-guide/code-conventions.md [hunk 1]
```diff
@@ -36,19 +36,7 @@ module.exports = {
...
-```js
-/**
- * @fileoverview Description of the file
- * @author Author's Name
- * @copyright 2015 Your Name. All rights reserved.
- * @copyright 2014 Author's Name. All rights reserved.
- * See LICENSE file in root directory for full license.
- */
-```
+The `@author` field gives you credit for having created the file. There is no need to assert copyright on individual files.
```
Comment (platinumazure) [inline docs/developer-guide/code-conventions.md]: Could use some help here on whether I should mention the jQuery license, or just remove the sentence about copyright altogether, or something else.

Comment (nzakas) [inline docs/developer-guide/code-conventions.md]: I think we can remove it.

**Comment by nzakas** [comment]
Just one small note. This doc was copied from my book and was focused on web deployments, that's why strict mode was the reverse of what we do. I guess further proof that no one reads this doc. :)

**Comment by platinumazure** [comment]
@nzakas Even if nobody else reads the doc, *I* do! :smile: I will make your suggested change and push tonight.

**Comment by platinumazure** [comment]
@nzakas Should be good to go now. Thanks for your patience.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**